cdap-source-control/src/main/java/io/cdap/cdap/sourcecontrol/RepositoryManagerFactory.java

```diff
@@ -29,14 +31,17 @@ public class RepositoryManagerFactory {
 
   private final SecureStore secureStore;
   private final CConfiguration cConf;
+  private final RepositoryStore repoStore;
 
   @Inject
-  RepositoryManagerFactory(SecureStore secureStore, CConfiguration cConf) {
+  RepositoryManagerFactory(SecureStore secureStore, CConfiguration cConf, RepositoryStore repoStore) {
```

Comment:
Will we have access to required tables by RepositoryStore from within taskworker ?

Reply:
No, that's the drawback. We need something like a `RepositoryConfigFetcher` and an internal endpoint. 

fyi
 @albertshau @tivv

Reply:
@albertshau yeah I agree. Currently we pass in the config and it's not a problem. And I've held off this PR.